tests/test_callback_for.py

```diff
@@ -0,0 +1,119 @@
+import scrapy
+import pytest
+from scrapy.utils.reqser import request_to_dict
+
+from scrapy_po.webpage import ItemPage, ItemWebPage
+from scrapy_po.utils import (
+    callback_for,
+    is_response_going_to_be_used,
+    DummyResponse,
+)
+
+
+class FakeItemPage(ItemPage):
+
+    def to_item(self):
+        return 'it works!'
+
+
+class FakeItemWebPage(ItemWebPage):
+
+    def to_item(self):
+        return 'it works!'
+
+
+class MySpider(scrapy.Spider):
+
+    name = 'my_spider'
+    parse_item = callback_for(FakeItemPage)
+    parse_web = callback_for(FakeItemWebPage)
+
+
+def test_callback_for():
+    """Simple test case to ensure it works as expected."""
+    cb = callback_for(FakeItemPage)
+    assert callable(cb)
+
+    fake_page = FakeItemPage()
+    response = DummyResponse('http://example.com/')
+    assert list(cb(response=response, page=fake_page)) == ['it works!']
+
+
+def test_callback_for_instance_method():
+    spider = MySpider()
+    response = DummyResponse('http://example.com/')
+    fake_page = FakeItemPage()
+    assert spider.parse_item(response, page=fake_page)
+
+
+def test_callback_for_inline():
+    callback = callback_for(FakeItemPage)
+    response = DummyResponse('http://example.com/')
+    fake_page = FakeItemPage()
+    assert callback(response, page=fake_page)
+
+
+def test_default_callback():
+    """Sample request not specifying callback."""
+    spider = MySpider()
+    request = scrapy.Request('http://example.com/')
+    request_dict = request_to_dict(request, spider)
+    assert isinstance(request_dict, dict)
```

Comment:
Is this required?
Asserting scrapy code..

Reply:
It's a public method. Also, that's a nice way to make sure a callback is serializable among its request.